storage/innobase/srv/srv0start.cc

```diff
@@ -324,8 +324,6 @@ static dberr_t create_log_file(bool create_new_db, lsn_t lsn,
 		srv_startup_is_before_trx_rollback_phase = false;
 	}
 
-	/* Enable checkpoints in buf_flush_page_cleaner(). */
-	recv_sys.recovery_on = false;
```

Comment:
This had better be replaced with `ut_ad(!recv_sys.recovery_on)` to avoid any surprise.